lib/elixir/lib/gen_event.ex

```diff
@@ -380,8 +380,23 @@ defmodule GenEvent do
         :gen.start(GenEvent, mode, @no_callback, [], [])
       atom when is_atom(atom) ->
         :gen.start(GenEvent, mode, {:local, atom}, @no_callback, [], [])
-      other when is_tuple(other) ->
-        :gen.start(GenEvent, mode, other, @no_callback, [], [])
+      {:global, term} = tuple when is_atom(term) ->
+        :gen.start(GenEvent, mode, tuple, @no_callback, [], [])
+      {:via, module, name} = tuple when is_atom(module) and is_atom(name) ->
```

Comment:
The `module` should respond to a `whereis_name` function that, in the tests is passing the variable `name` to `Process.whereis(name)` where it should be an `atom`.